unidist/core/backends/mpi/core/communication.py

```diff
@@ -338,6 +422,38 @@ def mpi_send_buffer(comm, buffer_size, buffer, dest_rank):
     comm.Send([buffer, MPI.CHAR], dest=dest_rank, tag=common.MPITag.BUFFER)
 
 
+def mpi_send_byte_buffer(comm, byte_array, dest_rank):
```

Comment:
As I see, we don't use mpi_send_buffer anymore. Maybe we should replace mpi_send_byte_buffer to mpi_send_buffer then. Same for mpi_recv_byte_buffer.